document kubernetes readiness pitfall (#2638)
fastify/fastify: docs/Recommendations.md

[scale-horiz]: https://en.wikipedia.org/wiki/Scalability#Horizontal
[why-use]: https://web.archive.org/web/20190821102906/https://medium.com/intrinsic/why-should-i-use-a-reverse-proxy-if-node-js-is-production-ready-5a079408b2ca
[haproxy]: https://www.haproxy.org/

> ## Kubernetes

Reviewer: The TOC needs to be updated on line 8.

> Please consider following points if you want to run fastify on kubernetes.
> 
> ### Readiness Probe
> 
> ```yaml
> readinessProbe:
>     httpGet:
>         path: /health
>         port: 4000
>     initialDelaySeconds: 30
>     periodSeconds: 30
>     timeoutSeconds: 3
>     successThreshold: 1
>     failureThreshold: 5
> ```
> 
> The `readinessProbe` use [(by default](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#configure-probes)) the pod IP as hostname. Fastify listen on `127.0.0.1` by default. Your probe won't be able to reach your application. In order to make it work you have to listen on `0.0.0.0` or specify a custom hostname in the `readinessProbe.httpGet` spec.

Reviewer: ```suggestion
The `readinessProbe` uses [(by default](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#configure-probes)) the pod IP as the hostname. Fastify listens on `127.0.0.1` by default. The probe will not be able to reach the application in this case. In order to make it work, the application must listen on `0.0.0.0` or specify a custom hostname in the `readinessProbe.httpGet` spec, as per the following example:

```yaml
readinessProbe:
    httpGet:
        path: /health
        port: 4000
    initialDelaySeconds: 30
    periodSeconds: 30
    timeoutSeconds: 3
    successThreshold: 1
    failureThreshold: 5
```
```

[scale-horiz]: https://en.wikipedia.org/wiki/Scalability#Horizontal
[why-use]: https://web.archive.org/web/20190821102906/https://medium.com/intrinsic/why-should-i-use-a-reverse-proxy-if-node-js-is-production-ready-5a079408b2ca
[haproxy]: https://www.haproxy.org/

> ## Kubernetes

Reviewer: ```suggestion
## Kubernetes
<a id="kubernetes"></a>
```
